src/content/blog/astro-570.mdx

```diff
@@ -0,0 +1,235 @@
+
+```ts title="src/pages/api/cart.ts" "session.load(sessionId)"
+export async function POST({ session, request }: APIContext) {
+  const sessionId = request.headers.get('x-session-id');
+
+  // If there is an existing session ID, load it. Otherwise one will be created
+  if (sessionId) {
+    await session.load(sessionId);
+  } 
+
+  const cart = await session.get('cart');
+  const data = await request.json<{ item: string }>();
+
+  if (!data?.item) {
+    return new Response('Item is required', { status: 400 });
+  }
+  session.set('cart', [...cart, data.item]);
+  // Return the cart and session ID
+  return Response.json({ cart, sessionId: session.sessionId });
+}
+```
+
+
+### Storage drivers
+
+Astro sessions use storage drivers to persist the data, and these are either zero-config or require minimal setup for all official Astro adapters. Alternatively you can use any [unstorage](https://unstorage.unjs.io/) driver, which supports dozens of storage backends such as Redis and MongoDB.
+
+- **[Node](https://docs.astro.build/en/guides/integrations-guide/node/#sessions)**: zero-config filesystem driver
+- **[Netlify](https://docs.astro.build/en/guides/integrations-guide/netlify/#sessions)**: zero-config Netlify Blobs driver
+- **[Cloudflare](https://docs.astro.build/en/guides/integrations-guide/cloudflare/#sessions)**: minimal-config Workers KV driver
+- **[Vercel](https://docs.astro.build/en/guides/integrations-guide/vercel/#sessions)**: minimal-config Redis/Upstash driver
+
+On any of these platforms you can easily switch out the driver for a different one if you don't want to use the default.
+
+```js title="astro.config.mjs" {5-10}
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  // ...
+  session: {
+    driver: 'redis',
+    options: {
+      url: process.env.REDIS_URL,
+    },
+  },
+});
+
+```
+
+### Upgrading from experimental sessions
+
+If you were using the experimental sessions API, you should remove the experimental flag from your `astro.config.mjs` file:
+
+```js title="astro.config.mjs" del={6}
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  // ...
+  experimental: {
+    session: false,
+  },
+});
+```
+Check out the [sessions guide](https://docs.astro.build/en/guides/sessions/) to learn more about how to use them in your Astro project.
+
+## SVG Components
+
+...
+
+## Config Imports
+
+The new virtual module `astro:config` is now stable!
+
+The virtual module allows to import a subset of the Astro configuration, with type-safety in mind. Two sub-paths are exposed for controlled access to your configuration:
+
+- `astro:config/client`: exposes config information that is safe to expose to the client.
+- `astro:config/server`: exposes additional information that is safe to expose to the server, such as file/dir paths.
+
+For example, you can now build your own path utilities without relying on an integrations:
```

Comment:
```suggestion
For example, you can now build your own path utilities without relying on an integration. Properties like `base` and `trailingSlash` set in your configuration can be imported and used in a `.js` file:
```
It just seemed like a long string of code samples without any indication what is happening, so I'm adding something to break them up.